Controllers/Frontend/Transparent.php

```diff
@@ -29,30 +30,34 @@ public function getWhitelistedCSRFActions()
      */
     public function redirectAction()
     {
-        $allowedPostParams = ['MD', 'PaRes'];
         $redirectUrl = Shopware()->Router()->assemble([
             'controller' => 'process',
             'action' => 'return',
         ]);
 
         $this->View()->assign('redirectUrl', $redirectUrl);
-        $this->View()->assign('postParams', $this->retrievePostParams($allowedPostParams));
+        $this->View()->assign('redirectParams', $this->retrieveParams());
         $this->logger->debug('Forward incoming POST response to process/return', [
-            'POST parameter keys' => $allowedPostParams
+            'POST and GET parameter keys' => self::ALLOWED_PARAMS
         ]);
     }
 
-    private function retrievePostParams(array $allowedParams): array
+    private function retrieveParams(): array
     {
-        $params = [];
-        foreach ($allowedParams as $key) {
-            if (null === $value = $this->Request()->getPost($key)) {
-                continue;
+        $request = $this->Request();
+
+        //Getting all GET parameters except for Shopware's action, controller and module
+        $getParams = $request->getQuery();
+        unset($getParams['action'], $getParams['controller'], $getParams['module']);
+
+        //Filtering allowed POST parameters
+        $fullPostParams = $request->getParams();
+        $postParams = [];
+        foreach (self::ALLOWED_PARAMS as $allowedParam) {
+            if (isset($fullPostParams[$allowedParam])) {
+                $postParams[$allowedParam] = $fullPostParams[$allowedParam];
             }
-
-            $params[$key] = $value;
         }
-
-        return $params;
+        return array_merge($getParams, $postParams);
```

Comment:
sorry to comment again :(
In case we send the get params as POST I think this will affect the process as well :(
The best would be to keep the get Params and compose the `$redirectUrl` using these as well, instead of converting the get to post.

Reply:
Currently all parameters are being received through POST here https://github.com/Adyen/adyen-shopware5/blob/develop/Controllers/Frontend/Process.php#L66, do you mean that we should change the receiving controller as well to interpret GET+POST?

Reply:
is this controller handling only 3DS1 or all the redirect payment methods redirect?
In case all of them then most of these return with GET and some parameters in the query.
To validate them we need to pass them to the /payments/details request but if we already do that then it's fine to do via POST as well here

Reply:
@cyattilakiss indeed, after the transparent controller forwards the GET and POST params the details call is done here https://github.com/Adyen/adyen-shopware5/blob/develop/Controllers/Frontend/Process.php#L69